src/Platforms/SQLitePlatform.php

```diff
@@ -917,7 +933,11 @@ private function getForeignKeysInAlteredTable(TableDiff $diff): array
                         ->getValue(),
                 );
 
-                $foreignKeys[$constraintKey] = $constraint;
+                if (! isset($keysByName[$constraintKey])) {
```

Comment:
Similarly. this should be `assert(sset($keysByName[$constraintKey]))`. This will allow to move `$foreignKeys[] = $constraint;` out of the `if` as suggested in [this comment](https://github.com/doctrine/dbal/pull/7158/files#r2366262152).